crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -297,12 +297,67 @@ impl<'db> SemanticIndexBuilder<'db> {
         }
     }
 
-    fn record_negated_constraint(&mut self, constraint: Constraint<'db>) {
+    fn add_constraint(&mut self, constraint: Constraint<'db>) -> Constraint<'db> {
+        self.current_use_def_map_mut().add_constraint(constraint);
+        constraint
+    }
+
+    fn add_negated_constraint(&mut self, constraint: Constraint<'db>) -> Constraint<'db> {
```

Comment:
It might be worth to add some documentation. E.g. to me it's unclear what the difference between `add_constraint` and `record_constraint` is. When should I use which?